### ISSUE TITLE
cinder e2e tests permafailing?

### ISSUE THREAD

**Issue by pierreprinetti** [kubernetes/cloud-provider-openstack#2220]
TL;DR: e2e tests in [this PR](https://github.com/kubernetes/cloud-provider-openstack/pull/2218) systematically fail.

While investigating with the help of @mdbooth, we found that the occm release manifests [are still referring to v1.27.0](https://github.com/kubernetes/cloud-provider-openstack/blob/52b4f44b7429c95be4f83fb2a68c5792747e9cb9/manifests/controller-manager/openstack-cloud-controller-manager-ds.yaml#L41). That version is broken IIUC, and that looks like a good root cause candidate.

One other thing that surprised me is that we apparently run release images as a base for presubmits in this job. Is that intentional? Shall we be running a CI build in the e2e CSI presubmit jobs?

CC @zetaab @mdbooth

**Comment by mdbooth** [comment]
So we definitely need to update the manifests to point to v1.27.1.

However, in terms of CI I personally think we should be running the code under test for all components in all jobs. So even though this is the csi-cinder job, I still think that occm should be running the code from the PR under test.

@zetaab If you agree with that either me or @pierreprinetti can knock up a PR to do that pretty quickly. What do you think?

**Comment by mdbooth** [comment]
I think #2221 should work round this, but as above I think the real problem is that we're running a release image at all.

**Comment by jichenjc** [comment]
#2221 merged, do we still need this to track or we can close this issue now?

**Comment by mdbooth** [comment]
@jichenjc Although #2221 worked round the problem, I am proposing that we fix the csi-cinder test to use an image built from the PR under test for *all* CPO components, not just CSI. It would be a very simple change. The only reason I haven't already submitted it already is that somebody deliberately created the current solution and I don't understand why.

**Comment by jichenjc** [comment]
>>>The only reason I haven't already submitted it already is that somebody deliberately created the current solution and I don't understand why.

I don't think someone deliberately do this ,if you have any thing to propose, please submit PR
